### PR TITLE
Redux in React components

### PR DESCRIPTION
**Pull request by Bhekisile**
## Changes

- Added initial state to booksSlice
- Dispatched actions and used useSelector to extract the state
- Created event handler for add and remove buttons

### CODE REVIEW

File: src/components/BookForm.js [hunk 1]
```diff
@@ -1,21 +1,32 @@
...
+        </select>
+        <button
+          className="btn"
+          type="submit"
+          onClick={() => {
+            dispatch(addBook);
+          }}
+        >
+          ADD BOOK
+        </button>
```
Comment (ShadyShawkat) [inline src/components/BookForm.js]: - [ ] Clicking this button refreshes the page and doesn't add a book as the page is reloaded. Kindly fix this behavior.

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @Bhekisile, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/BookForm.js [hunk 2]
```diff
@@ -1,21 +1,69 @@
 import '../styles/BookForm.css';
+import { useDispatch, useSelector } from 'react-redux';
+import { useState } from 'react';
+// import PropTypes from 'prop-types';
```
Comment (levy002) [inline src/components/BookForm.js:4]: - [OPTIONAL] KIndly make sure to remove all commented code in order to keep the code clean and professional.

**Review by levy002 — APPROVED** [review]
## Hello @Bhekisile 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- No linter error :heavy_check_mark: 
- Implemented the suggested changes :heavy_check_mark: 

## Status: APPROVED :fire:  :fire: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/BookForm.js (hunk 2 above)
Comment (Bhekisile) [reply src/components/BookForm.js:4]: I will do so, thank you.

**Pull request merged**